project/Boilerplate.scala

```diff
@@ -349,6 +349,9 @@ object Boilerplate {
         -  /** @group ParMapArity */
         -  def parMap$arity[M[_], ${`A..N`}, Z]($fparams)(f: (${`A..N`}) => Z)(implicit p: NonEmptyParallel[M]): M[Z] =
         -    p.flatMap.map(${nestedExpansion.products}) { case ${nestedExpansion.`(a..n)`} => f(${`a..n`}) }
+        -
+        -  def flatParMap$arity[M[_], ${`A..N`}, Z]($fparams)(f: (${`A..N`}) => M[Z])(implicit p: NonEmptyParallel[M]): M[Z] =
+        -    p.flatMap.flatten(p.flatMap.map(${nestedExpansion.products}) { case ${nestedExpansion.`(a..n)`} => f(${`a..n`}) })
```

Comment:
Not directly about this PR, but I'm confused by the code here. Why is it calling `p.flatMap.map` instead of `p.apply.map`? Since the `Apply` is the parallel instance.

Am I confused or is this a bug? Maybe we need some tests using e.g. `EitherNec[Throwable, A]`?

Reply:
Yes agree, good catch... So now I wonder too.

Reply:
Ahh, no this is completely fine. It's because it's calling vanilla `map` on a single argument (a tuple), which is then unpacked by the `case` in the lambda. I was mistakenly reading it as calling a `mapN`-like method, in which case this would be wrong.

The `map` must be consistent between the `flatMap` and the `apply`, and using the `apply` instance would require a conversion to the parallel datatype and back.

Reply:
Actually ... this is just `flatMap` ... right? 😂 

```suggestion
        -    p.flatMap.map(${nestedExpansion.products}) { case ${nestedExpansion.`(a..n)`} => f(${`a..n`}) }
        -
        -  def flatParMap$arity[M[_], ${`A..N`}, Z]($fparams)(f: (${`A..N`}) => M[Z])(implicit p: NonEmptyParallel[M]): M[Z] =
        -    p.flatMap.flatMap(${nestedExpansion.products}) { case ${nestedExpansion.`(a..n)`} => f(${`a..n`}) }
```

Reply:
It is! I run the tests locally just to be sure that I was not missing something syntax related :)